### PR TITLE
add try/except when for duplicate books

### PR DESCRIPTION
**Pull request by sunnycake**
@whagestad I added a try/except block to add_book() function. Can you verify if all looks good? Thanks.

### CODE REVIEW

**Comment by claraj** [comment]
I'm sure Will is also going to remark on this, but it looks like there's other code in this commit as well as the try-except. When you make branches, make sure you check out master first and then make the new branch from master.

**Comment by sunnycake** [comment]
I'm not sure why that's there also. I did branch it from the master and I
don't see the extra codes here on my end. Even the spacing on my codes
are different then what was pushed to github. Any idea why it's doing
that so I can fix it? Should I try to re-commit or wait until Will reviews it then commit again?

File: main.py [hunk 1]
```diff
@@ -52,19 +56,20 @@ def show_all_books():
 
 
 def search_book():
-    search_term = ui.ask_question('Enter search term, will match partial authors or titles.')
+    search_term = ui.ask_question(
```
Comment (claraj) [inline main.py:59]: This is a minor change but it's unrelated to the specific issue.

File: main.py [hunk 2]
```diff
@@ -52,19 +56,20 @@ def show_all_books():
...
     ui.show_books(matches)
 
 
 def change_read():
 
     book_id = ui.get_book_id()
-    book = store.get_book_by_id(book_id)  
-    new_read = ui.get_read_value()     
-    book.read = new_read 
+    book = store.get_book_by_id(book_id)
```
Comment (claraj) [inline main.py:67]: whitespace changes add clutter - try to avoid.  if there's issues with whitespace deal with it under another issue, branch, commit(s)

**Review by claraj — CHANGES_REQUESTED** [review]

File: main.py (hunk 2 above)
Comment (sunnycake) [reply main.py:67]: I will update this one under a different branch.

File: main.py (hunk 1 above)
Comment (sunnycake) [reply main.py:59]: I noticed that when I click ctrl+s the spacing automatically changes for this one. Is there a way to update this in vscode?

**Comment by claraj** [comment]
Take a look at the note in D2L - you can hide whitespace changes, like differences in spaces at the end of lines - in the diff settings at GitHub.
________________________________

**Review by whagestad — APPROVED** [review]
This looks good! Testing the code I found no issues with this function, worked as intended when trying to add a duplicate book.

**Comment by whagestad** [comment]
See lines 69-73 in the "Resolve Conflicts" link

Looking at the code it would appear the text is just replacing itself. Do you mind checking it out and then go ahead with merging once it's resolved?

**Pull request merged**